### PR TITLE
Change method override inference to use combined member signature

### PR DESCRIPTION
**Pull request by eernstg**

### CODE REVIEW

File: resources/type-system/inference.md [hunk 1]
```diff
@@ -150,22 +153,27 @@ error if there is such an ordering.  Note that method override inference is
...
+
+A compile-time error occurs if `s` does not exist.  *E.g., one
+superinterface could have signature `void m([int])` and another one could
+have signature `void m(num)`, such that none of them is most specific.
+There may still exist a valid override of both (e.g., `void m([num])`).  In
+this situation `C.m` can be declared with a complete signature, it just
+cannot use override inference.*
+
+If there is no corresponding parameter in `s` for a parameter of the
+declaration of `m` in `C`, it is treated as `dynamic` (*e.g., this occurs
```
Comment (eernstg) [inline resources/type-system/inference.md:174]: Maybe `Object?` rather than `dynamic`?

Comment (lrhn) [reply resources/type-system/inference.md:174]: Preferably, if possible.
Means that all existing code which writes nothing and inherits nothing must be migrated to an explicit `dynamic`. That's a good thing.

File: resources/type-system/inference.md [hunk 2]
```diff
@@ -150,22 +153,27 @@ error if there is such an ordering.  Note that method override inference is
...
-is being inferred for the overriding method, it is an error.  If there is no
-corresponding parameter position in the overridden method to infer from and the
-signatures are compatible, it is treated as dynamic (e.g. overriding a one
-parameter method with a method that takes a second optional parameter).  Note:
-if there is no corresponding parameter position in the overridden method to
-infer from and the signatures are incompatible (e.g. overriding a one parameter
-method with a method that takes a second non-optional parameter), the inference
-result is not defined and tools are free to either emit an error, or to defer
-the error to override checking.
+A method `m` of a class `C` which is subject to override inference is
```
Comment (leafpetersen) [inline resources/type-system/inference.md:159]: Maybe "A method `m` of a class `C` is subject to override inference if it"

File: resources/type-system/inference.md (hunk 1 above)
Comment (leafpetersen) [reply resources/type-system/inference.md:174]: I think we probably should land this specifying `dynamic`.  We can discuss a global no implicit dynamic change, but I honestly think it's too much to bite off this late in the game.

File: resources/type-system/inference.md [hunk 3]
```diff
@@ -150,22 +153,27 @@ error if there is such an ordering.  Note that method override inference is
...
+this situation `C.m` can be declared with a complete signature, it just
+cannot use override inference.*
+
+If there is no corresponding parameter in `s` for a parameter of the
+declaration of `m` in `C`, it is treated as `dynamic` (*e.g., this occurs
+when overriding a one parameter method with a method that takes a second
+optional parameter*).
 
 
 ##### Instance field, getter, and setter override inference
```
Comment (leafpetersen) [inline resources/type-system/inference.md:179]: Should this section be updated?  I think it should, right?  There is a notion of "combined member signature" for getters/setters/fields as well, right?

**Comment by leafpetersen** [comment]
What's the status of this?

File: resources/type-system/inference.md (hunk 2 above)
Comment (eernstg) [reply resources/type-system/inference.md:159]: Done.

File: resources/type-system/inference.md (hunk 1 above)
Comment (eernstg) [reply resources/type-system/inference.md:174]: Keeping `dynamic`.

File: resources/type-system/inference.md (hunk 3 above)
Comment (eernstg) [reply resources/type-system/inference.md:179]: Done.

**Comment by eernstg** [comment]
Updated the getters/setters section, PTAL.

**Pull request merged**